### ISSUE TITLE
Game will sometime end up in a locked state 

### ISSUE THREAD

**Issue by nguyen-jhu** [JHU-601/flask-production#56]
Inconsistent, but every so often though a player's turn is stated in the corner, the player cannot make moves or anything else.

**Comment by KenDJohnson** [comment]
Is this different from #53? That will do that too.

**Comment by nguyen-jhu** [comment]
Ken, this differs from #53. Happens without the usage of suggest. I believe I have encountered where this may be going wrong.

Given the situation:
- Player 1 registers before anyone enters lobby (Known issue is that the display will not show who is in the lobby for subsequent players - but can still start the game)
- Start Game
- Player 1 and Player 2 screen shows "Your Turn", with all others showing "Player 2's turn".
- Updates differently between player 1 and all other players for "turn notification" - off by 1 player 
- Upon an error state - like a invalid move - game will end up in a gridlock

**Comment by KenDJohnson** [comment]
> Player 1 registers before anyone enters lobby (Known issue is that the display will not show who is in the lobby for subsequent players - but can still start the game)


Wait, what? That shouldn't be possible. The server is responsible for starting the game, and it won't start it until everyone is registered.
Also not sure I'm following the rest of this. Could you write up how to repro this later? Or just push up a fix if you think you know what's going on.

**Comment by stephengrice** [comment]
I think I know what this is. I noticed the bug, should be fixed by #51.

This was an issue with how players were being looked-up on the clientside. When processing messages related to whose turn it was, the player was looked up by the character id from the server in the **unsorted** list of characters. As a result, the client thought it was another player's turn when it was really theirs.

This, and at least one other bug, did not show up when we ran our Selenium script because it has all players join and register in a perfect order. I would expect that since he was doing it by hand, Phuong may have registered users in a different order, uncovering the bug.

That being said, I think we can probably close. Maybe Phuong can confirm that the most recent commit in develop does not produce this issue. I can also re-create it locally if you'd like.

The problem was on [this line](https://github.com/JHU-601/flask-production/blob/develop/clueless/static/js/controllers.js#L281). 

```javascript
var player = this.gameState.players[message.player]; // does not use the ugly for loop that was necessary before
```

But the refactored fix on [this line](https://github.com/JHU-601/flask-production/blob/develop/clueless/static/js/controllers.js#L6) (and a few other lines) eliminates the issue.

```javascript
this.players = [null, null, null, null, null, null]; // look up player by character-id rather than arbitrary join order
```
